lib/sidebar.rb

```diff
@@ -29,185 +29,193 @@ def pre_render(page)
+          sub_group_markup << '<ul class="nav-ul">'
+
+          if grandchildren.empty?
+            headers.each do |header|
+              hash = header[:hash]
+              subtitle = header[:title]
+              sub_group_markup << "<li class=\"nav-leaf\"><a href=\"#{url}#{hash}\">#{subtitle}</a></li>"
             end
           else
-            subgroup_leaf_class = active ? 'nav-leaf nav-subgroup-leaf active' : 'nav-leaf nav-subgroup-leaf'
-            subgroup << "<li class=\"#{subgroup_leaf_class}\"><a href=\"#{value[:url]}\">#{title} overview</a></li>"
+            sub_group_leaf_class = active ? 'nav-leaf nav-subgroup-leaf active' : 'nav-leaf nav-subgroup-leaf'
+            sub_group_markup << "<li class=\"#{sub_group_leaf_class}\"><a href=\"#{url}\">#{title} overview</a></li>"
 
-            subsubgroup_list.each do |subsubgroup_key, subsubgroup_value|
-              subgroup << generateSubgroup(filename, subsubgroup_key, subsubgroup_value, subsubgroup_list, 3)
+            grandchildren.each do |grandchild_path, grandchild_page|
+              sub_group_markup << generate_sub_group(filename, grandchild_path, grandchild_page, grandchildren, 3)
             end
           end
 
-          subgroup << '</ul>'
-          subgroup << '</li>'
+          sub_group_markup << '</ul>'
+          sub_group_markup << '</li>'
         else
-          value[:headers].each do |header|
-            subgroup << "<li class=\"nav-leaf\"><a href=\"#{value[:url]}#{header[:hash]}\">#{header[:title]}</a></li>"
+          headers.each do |header|
+            hash = header[:hash]
+            subtitle = header[:title]
+            sub_group_markup << "<li class=\"nav-leaf\"><a href=\"#{url}#{hash}\">#{subtitle}</a></li>"
           end
         end
       else
-        subgroup << if has_subgroups
-                      "<li class=\"nav-leaf nav-subgroup-leaf\"><a href=\"#{value[:url]}\">#{title}</a></li>"
-                    else
-                      "<li class=\"nav-leaf\"><a href=\"#{value[:url]}\">#{title}</a></li>"
-                    end
+        sub_group_markup << if has_child_pages
+                              "<li class=\"nav-leaf nav-subgroup-leaf\"><a href=\"#{url}\">#{title}</a></li>"
+                            else
+                              "<li class=\"nav-leaf\"><a href=\"#{url}\">#{title}</a></li>"
+                            end
       end
 
-      subgroup
+      sub_group_markup
     end
 
     def render(filename)
-      sidebar = ''
+      sidebar_markup = ''
+      pages = @pages_pre_render.safe_merge(@pages_with_headers).sort_by { |_, page| page[:menu_order] }
 
-      merged = merge(@hash_pre_render, @filename_with_headers).sort_by { |_key, value| value[:menu_order] }
-      merged.select { |key, _value| key.split('/').length <= 2 }.each do |key, value|
-        next if value[:title].nil?
-        next if value[:hide_from_sidebar]
+      pages.select { |path, _| path.split('/').length <= 2 }.each do |path, page|
```

Comment:
Codacy found an issue: [Line is too long. [81/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)